CMakeLists.txt

```diff
@@ -61,7 +77,7 @@ create_translation(engen_qm_file ${CMAKE_BINARY_DIR}/sailfish-office.ts ${sailfi
 
 add_executable(sailfish-office ${sailfishoffice_SRCS} ${engen_qm_file})
 qt5_use_modules(sailfish-office Widgets Quick DBus)
-target_link_libraries(sailfish-office stdc++ ${QT_LIBRARIES} ${BOOSTER_LIBRARY} ${QTSPARQL_LIBRARY})
+target_link_libraries(sailfish-office stdc++ uchardet ${QT_LIBRARIES} ${BOOSTER_LIBRARY} ${QTSPARQL_LIBRARY})
```

Comment:
Is uchardet really linked to the main sailfish-office library ? Is it not required just for the text plugin ?

Reply:
No. I've answered, that it is a separated library and was not linked. I gave sources to Pekka for review

Reply:
Yes, I know, I've read the history of the excahnge, my question is : is it necessary to link uchardet to sailfish-office library when it is used only by the text plugin ?

Reply:
Yes because it will be also used in calligra to find charset for csv files

Reply:
I see, but excuse my insistance, should it be linked by the calligra library in that case (or by the calligra plugin if the charset detection is done before passing the data to calligra itself) ?

Reply:
Yes, it should. Previosly when a user opens a CSV file, a dialog (developed via QWidgets) appears where user should choose encoding and delimiter himself. 
Now we came to a conclusion with Pekka that this dialog should be removed and a csv file should be opened with an original encoding. Thats why uchardet detecting in calligra for csv files should be necessary.

Reply:
I understand the necessity and concur with the requirement for uchardet, but in my opinion, this is the wrong place to link with uchardet for calligra. The link should be done in calligra project itself since it's where the actual code using uchardet will be.

From my understanding:
- Calligra is used via QML components in sailfish office, so it means that the libcalligracomponent.so is dlopened by the QML engine when a new calligra object is required from QML.
- This calligra component will use uchardet and thus requires to resolve some symbols like `uchardet_handle_data`. If libcalligracomponent.so is not dependent on uchardet, the resolving will work only if uchardet_handle_data is already in memory via a previous dlopen of another library. If you link it with sailfish-office this will actually be the case and everything will work.
- But if you dlopen libcalligracomponent.so from another software for any reason, it will fail with unresolved symbols because of the missing linkage with uchardet library.

That's why I'm suggesting not to link uchardet with sailfish-office library but with the library that is actually using the code and I don't see here that uchardet is used to decode CSV files.

Reply:
It's not used here to decode CSV files. What about QML...Actually QML part of sailfish-office doesn't know about Calligra a lot. We just set the source file and then wait until it'll be displayed. Sailfish-office doesn't know anything about the internal processes when we call "setSource" method in calligra. We need uchardet for the internal part.

What about using uchardet - actually it is linked in cmakelists in "CSV" folder and is used for csv

Reply:
> Yes because it will be also used in calligra to find charset for csv files

This will be up to calligra folks to consider. May or may not happen.

In general Damien has a point, module (being it library or binary) that uses some other library should link against that, others don't need to care. If main application doesn't explicitly use uchardet, this should be dropped.